src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,49 +1,140 @@
 package core.basesyntax;
 
+import edu.emory.mathcs.backport.java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    Object [] elements;
+    private int size = 0;
+
+    public ArrayList() {
+        this.elements = new Object[DEFAULT_CAPACITY];
+    }
 
     @Override
     public void add(T value) {
-
+        add(value, size);
     }
 
     @Override
     public void add(T value, int index) {
-
+        checkRange(index);
+        if (elements.length == size) {
+            increaseArraySize();
+        }
+        for (int i = size; i > index; i--) {
+            elements[i] = elements[i - 1];
+        }
+        elements[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        T[] listData = (T[]) new Object[list.size()];
+        for (int i = 0; i < listData.length; i++) {
+            listData[i] = list.get(i);
+        }
+        System.arraycopy(listData,0, elements,size, listData.length);
+        size += listData.length;
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        return (T) elements[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        checkIndex(index);
+        elements[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkRange(index);
+        T removed = (T) elements[index];
+        for (int i = index; i < size; i++) {
+            elements[i] = elements[i + 1];
+        }
+        size--;
+        return removed;
     }
 
     @Override
     public T remove(T t) {
-        return null;
+        int index = getIndexElement(t);
+
+        if (index == -1) {
+            return null;
+        }
+        if (isExist(t)) {
```

Comment:
if getIndex() gave us a proper index(not minus 1) should we check one more time?

Reply:
re-write!